src/platform/platform_posix.c

```diff
@@ -771,13 +858,26 @@ CxPlatThreadWait(
     CXPLAT_FRE_ASSERT(pthread_join(*Thread, NULL) == 0);
 }
 
-uint32_t
+CXPLAT_THREAD_ID
 CxPlatCurThreadID(
     void
     )
 {
-    CXPLAT_STATIC_ASSERT(sizeof(pid_t) <= sizeof(uint32_t), "PID size exceeds the expected size");
-    return syscall(__NR_gettid);
+
+#if defined(CX_PLATFORM_LINUX)
+
+    CXPLAT_STATIC_ASSERT(sizeof(pid_t) <= sizeof(CXPLAT_THREAD_ID), "PID size exceeds the expected size");
+    return gettid();
```

Comment:
Why change this from what it was?

Reply:
Because we shouldn't be using syscalls directly, and especially not the NR versions of the syscalls. gettid returns the exact same value, but actually in a posix compatible way.